README.md

```diff
@@ -16,39 +16,137 @@ and to keep them as similar as possible:
+as a normal user as long as `/dev/kvm` is accessible (a prerequisite
+mentioned above) and thus we are able to perform a compose as a normal
+user. This allows us to run inside of a locked down OpenShift
+environment, which is where we are running our builds for Fedora
+CoreOS currently. 
+
+We recommend you use unprivileged mode when building locally if you
+are hacking on Fedora CoreOS so that you can mimic our build
+environment as much as possible.
+
+#### Privileged Mode
+
+In privileged mode the rpm-ostree compose uses container features
+itself, thus requires a privileged container in order to perform the
+compose. This is known as [recursive containers](https://github.com/projectatomic/bubblewrap/issues/284).
+
+### Container Build
 ---
 
-Here we store data in `/srv/coreos` on our host system.  You can choose
-any directory you like.  You should run these commands as `root`.
+To completely rebuild the coreos-assembler container image locally, execute
+`$ sudo podman build -t localhost/coreos-assembler .` or
+`$ sudo podman build -t localhost/coreos-assembler -f Dockerfile.rhel .`
+from the `coreos-assembler` repository. If building the RHEL version please 
+note you will need a maipo directory with references to
+proper repositories.
+
+You can optionally upload your built image to a registry such as quay.io by
+doing the following:
 
 ```
-$ mkdir /srv/coreos
-$ cd /srv/coreos
-$ alias coreos-assembler='podman run --rm -ti --privileged --userns=host -v $(pwd):/srv --workdir /srv quay.io/coreos-assembler/coreos-assembler'
+$ sudo podman push <image id> example.com/<account name>/coreos-assembler
 ```
 
-If you need access to CA certificates on your host (for example, when you need to access
-a git repo that is not on the public Internet), you can mount in the host certificates
-as read-only.  For example, on a Fedora host the alias would change to:
+### Setup
+---
 
-`$ alias coreos-assembler='podman run --rm -ti --privileged --userns=host -v /etc/pki:/etc/pki:ro -v $(pwd):/srv --workdir /srv quay.io/coreos-assembler/coreos-assembler'`
+Let's set up our working directory first. We'll create and use `./srv-coreos`
+on our host system. You can choose any directory you like. 
 
-See this [Stack Overflow question](https://stackoverflow.com/questions/26028971/docker-container-ssl-certificates) for additional discussion.
+```
+$ mkdir ./srv-coreos
+$ setfacl -m u:1000:rwx ./srv-coreos
+$ setfacl -d -m u:1000:rwx ./srv-coreos
+$ chcon system_u:object_r:container_file_t:s0 ./srv-coreos
+$ cd ./srv-coreos
+```
+
+In the above commands we:
+
+- created the `./srv-coreos` directory
+- set file ACLs so that the `builder` user (uid `1000` inside the container) can write files
+- set file ACLs so that new files get created with ACLs that allow the `builder` user
+- gave the directory an SELinux file context for sharing with containers
+- changed our working directory into `./srv-coreos`
```

Comment:
Minor/optional: I'd just make these `# comment lines` in the previous block to make it easier to follow.